### PR TITLE
CDAP-15641 Spanner integration tests

### PR DESCRIPTION
**Pull request by vladhlinskiy**
JIRA: https://issues.cask.co/browse/CDAP-15641

In the scope of this PR:
* Created Spanner integration tests

### CODE REVIEW

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleCloudSpannerTest.java [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+
+  @Override
+  protected void innerSetup() throws Exception {
+    Tasks.waitFor(true, () -> {
+      try {
+        final ArtifactId dataPipelineId = TEST_NAMESPACE.artifact("cdap-data-pipeline", version);
+        if (!isSpannerPluginExists(dataPipelineId, BatchSource.PLUGIN_TYPE)) {
+          return false;
+        }
+        return isSpannerPluginExists(dataPipelineId, BatchSink.PLUGIN_TYPE);
```
Comment (mlozbin-cybervisiontech) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleCloudSpannerTest.java]: return isSpannerPluginExists(dataPipelineId, BatchSource.PLUGIN_TYPE))  && isSpannerPluginExists(dataPipelineId, BatchSink.PLUGIN_TYPE)) ?

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleCloudSpannerTest.java [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+      .set("TIMESTAMP_COL").to(Timestamp.ofTimeSecondsAndNanos(NOW.toEpochSecond(), NOW.getNano()))
+      .set("NOT_IN_THE_SCHEMA_COL").to("some string")
+      .build()
+  );
+
+  private static final List<Mutation> SOURCE_TABLE_TEST_MUTATIONS = TEST_MUTATIONS.apply(SOURCE_TABLE_NAME);
+
+  private static final BiConsumer<Iterator<Value>, ResultSet> COMMON_TEST_ACTIONS = (expectedValues, resultSet) -> {
+    Assert.assertEquals(expectedValues.next().getInt64(), resultSet.getLong("ID"));
+    if (expectedValues.hasNext()) {
```
Comment (mlozbin-cybervisiontech) [inline integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleCloudSpannerTest.java]: please avoid conditions in tests

**Review by mlozbin-cybervisiontech — CHANGES_REQUESTED** [review]

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleCloudSpannerTest.java (hunk 2 above)
Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleCloudSpannerTest.java]: Changed to: 
```
  private void verifySinkData(ResultSet resultSet) {
    Assert.assertTrue(resultSet.next());
    Map<String, Value> firstRowExpected = SOURCE_TABLE_TEST_MUTATIONS.get(0).asMap();
    Assert.assertEquals(firstRowExpected.get("ID").getInt64(), resultSet.getLong("ID"));

    Assert.assertTrue(resultSet.next());
    Map<String, Value> secondRowExpected = SOURCE_TABLE_TEST_MUTATIONS.get(1).asMap();
    Assert.assertEquals(secondRowExpected.get("ID").getInt64(), resultSet.getLong("ID"));
    Assert.assertEquals(secondRowExpected.get("STRING_COL").getString(), resultSet.getString("STRING_COL"));
    Assert.assertEquals(secondRowExpected.get("BOOL_COL").getBool(), resultSet.getBoolean("BOOL_COL"));
    Assert.assertEquals(secondRowExpected.get("BYTES_COL").getBytes(), resultSet.getBytes("BYTES_COL"));
    Assert.assertEquals(secondRowExpected.get("DATE_COL").getDate(), resultSet.getDate("DATE_COL"));
    Assert.assertEquals(secondRowExpected.get("FLOAT_COL").getFloat64(), resultSet.getDouble("FLOAT_COL"), 0.00001);
    Assert.assertEquals(secondRowExpected.get("TIMESTAMP_COL").getTimestamp(), resultSet.getTimestamp("TIMESTAMP_COL"));
  }
```

File: integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleCloudSpannerTest.java (hunk 1 above)
Comment (vladhlinskiy) [reply integration-test-remote/src/test/java/io/cdap/cdap/app/etl/batch/GoogleCloudSpannerTest.java]: Changed. Thanks.

**Pull request merged**